include/library.h

```diff
@@ -34,6 +34,14 @@
 
 #define KIWIX_LIBRARY_VERSION "20110515"
 
+#define NODISCARD
+#if defined __has_cpp_attribute
+  #if __has_attribute (nodiscard)
```

Comment:
Shouldn't this be `__has_cpp_attribute` for better portability (and consistency with the above check)?

Reply:
Yes